### DISCUSSION TITLE
Issue with Symbol Visibility in net.unix Library

### DISCUSSION THREAD

**Discussion by GGRei** [vlang/v#20174] [Questions and Answers]
Hello,

In my current project on a Windows environment ( 10 ), I have encountered an issue with the net.unix library.

The issue arises even when simply importing the library into the project, without writing any additional code. This leads to errors related to symbol visibility with constants used by the library :

### The errors :

**1- v/vlib/net/unix/stream.c.v:73:54: error: constant 'net.msg_nosignal' is private**
`C.photon_send(c.sock.handle, ptr, remaining, net.msg_nosignal, c.write_timeout)`

**2- v/vlib/net/unix/stream.c.v:75:47: error: constant 'net.msg_nosignal' is private**
`C.send(c.sock.handle, ptr, remaining, net.msg_nosignal)`

**3- v/vlib/net/unix/stream.c.v:380:47: error: constant 'net.fionbio' is private**
`net.socket_error(C.ioctlsocket(handle, net.fionbio, &t))!`

**4- v/vlib/net/unix/stream.c.v:493:47: error: constant 'net.fionbio' is private**
`net.socket_error(C.ioctlsocket(sockfd, net.fionbio, &t))!`

### Test code to replicate the error:

```
module main

import net.unix

fn main() {
	println("test")
}
```

I have not yet had the opportunity to test this on Ubuntu and a review of the standard library documentation has not yielded any relevant information to resolve this issue.

Could this be an issue with how I am using the library ( My development environment is not suited for this library? Need to write a wrapper of '**Named Pipes**'? ), or is it a potential problem in the net.unix library itself?

Thank you for your assistance.

**Comment by GGRei** [comment]
PR submitted : [net: fix windows constants to make them public for net.unix usage](https://github.com/vlang/v/pull/20183)

**Comment by shove70** [comment]
Ok, thanks
